### PR TITLE
kicad: update to 6.0.1

### PR DESCRIPTION
**Pull request by PoroCYon**
#### Testing the changes
- I tested the changes in this PR: **YES**/**briefly**: I opened a project of mine, looked around and made some minor edits, tried the main components (schematic editor, PCB editor, 3D viewer), but I didn't test *all* features (scripting, gerber output with weird contents, SPICE integration, ...).

[ci skip] because `kicad-packages3D` is humongous, and KiCAD itself takes a long while to build, and all other packages are trivial (only assets).

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I did **not** build this PR locally for these architectures as I don't have any of these machines (thus cannot test it) and cross-compiling would take too long.

So, preferably, a few other people should test this PR as well.

### CODE REVIEW

**Comment by unspecd** [comment]
1. It works for me (on x86_64-musl), but one test failed:
```
=> kicad-6.0.0_1: running do_check ...
[0/1] Running tests...
Test project /builddir/kicad-6.0.0/build
    Start 1: qa_python
1/8 Test #1: qa_python ........................   Passed   63.83 sec
    Start 2: qa_common
2/8 Test #2: qa_common ........................   Passed    2.34 sec
    Start 3: qa_gerbview
3/8 Test #3: qa_gerbview ......................   Passed   21.09 sec
    Start 4: qa_eeschema
4/8 Test #4: qa_eeschema ......................***Failed   15.42 sec
Running 62 test cases...
../qa/eeschema/test_ee_item.cpp(124): error: in "EeItem/Rotate": check aItem->GetBoundingBox().GetTop() == aOriginalItem->GetBoundingBox().GetTop() has failed [-529945 != -529188]
Failure occurred in a following context:
    Class: SCH_SHEET
    Position: -1000000 -500000, Reference: -1000000 -500000
../qa/eeschema/test_ee_item.cpp(128): error: in "EeItem/Rotate": check aItem->GetBoundingBox().GetBottom() == aOriginalItem->GetBoundingBox().GetBottom() has failed [-433225 != -433982]
Failure occurred in a following context:
    Class: SCH_SHEET
    Position: -1000000 -500000, Reference: -1000000 -500000
../qa/eeschema/test_ee_item.cpp(124): error: in "EeItem/Rotate": check aItem->GetBoundingBox().GetTop() == aOriginalItem->GetBoundingBox().GetTop() has failed [-529945 != -529188]
Failure occurred in a following context:
    Class: SCH_SHEET
    Position: -1000000 -500000, Reference: -1000000 0
../qa/eeschema/test_ee_item.cpp(128): error: in "EeItem/Rotate": check aItem->GetBoundingBox().GetBottom() == aOriginalItem->GetBoundingBox().GetBottom() has failed [-433225 != -433982]
Failure occurred in a following context:
    Class: SCH_SHEET
    Position: -1000000 -500000, Reference: -1000000 0

[etc...]

*** 162 failures are detected in the test module "Common Eeschema module tests"


    Start 5: qa_sexpr
5/8 Test #5: qa_sexpr .........................   Passed    0.13 sec
    Start 6: qa_kimath
6/8 Test #6: qa_kimath ........................   Passed    0.92 sec
    Start 7: qa_pcbnew
7/8 Test #7: qa_pcbnew ........................   Passed  125.88 sec
    Start 8: qa_kicad2step
8/8 Test #8: qa_kicad2step ....................   Passed    4.92 sec

88% tests passed, 1 tests failed out of 8

Total Test time (real) = 234.73 sec

The following tests FAILED:
	  4 - qa_eeschema (Failed)
Errors while running CTest
FAILED: CMakeFiles/test.util 
cd /builddir/kicad-6.0.0/build && /usr/bin/ctest --force-new-ctest-process
ninja: build stopped: subcommand failed.
```
2. You could remove unused configure_args:
   - KICAD_USE_OCE [[1](https://gitlab.com/kicad/code/kicad/-/commit/e57bde65821d7afcafeceb3b1a7f25e424d83121), [2](https://gitlab.com/kicad/code/kicad/-/commit/1c5c052301b607a0cf8b8ed6681f1db1622a865b)]
   - [KICAD_BUILD_VERSION](https://gitlab.com/kicad/code/kicad/-/commit/fd54d394bd12d2cb277fc289b6b4879f986c2416)
   - [BUILD_GITHUB_PLUGIN](https://gitlab.com/kicad/code/kicad/-/commit/9ed6cdd94391bf07b4d509b0b79f8873ec1ec4fd)
3. Not sure, but instead of deleting `disable-warnings.patch`, you could update it:
```diff
--- a/CMakeModules/Warnings.cmake
+++ b/CMakeModules/Warnings.cmake
@@ -38,13 +38,6 @@
     set( WARN_FLAGS_CXX "-Wall" )
 
 
-    # Warn about missing override specifiers
-    CHECK_CXX_COMPILER_FLAG( "-Wsuggest-override" COMPILER_SUPPORTS_WSUGGEST_OVERRIDE )
-
-    if( COMPILER_SUPPORTS_WSUGGEST_OVERRIDE )
-        set( WARN_FLAGS_CXX "${WARN_FLAGS_CXX} -Wsuggest-override" )
-        message( STATUS "Enabling warning -Wsuggest-override" )
-    endif()
```

**Comment by PoroCYon** [comment]
Thanks for the feedback, I've incorporated your suggestions. However, I can't seem to replicate  the test failure:

```
=> kicad-6.0.0_1: running do_check ...
[0/1] Running tests...
Test project /builddir/kicad-6.0.0/build
    Start 1: qa_python
1/8 Test #1: qa_python ........................   Passed    1.37 sec
    Start 2: qa_common
2/8 Test #2: qa_common ........................   Passed    0.04 sec
    Start 3: qa_gerbview
3/8 Test #3: qa_gerbview ......................   Passed    0.11 sec
    Start 4: qa_eeschema
4/8 Test #4: qa_eeschema ......................   Passed    0.83 sec
    Start 5: qa_sexpr
5/8 Test #5: qa_sexpr .........................   Passed    0.01 sec
    Start 6: qa_kimath
6/8 Test #6: qa_kimath ........................   Passed    0.02 sec
    Start 7: qa_pcbnew
7/8 Test #7: qa_pcbnew ........................   Passed   45.37 sec
    Start 8: qa_kicad2step
8/8 Test #8: qa_kicad2step ....................   Passed    0.17 sec

100% tests passed, 0 tests failed out of 8

Total Test time (real) =  47.93 sec
```

Maybe it's a glibc vs musl difference? I forgot to mention I'm using glibc. Either way, I'd need some help to fix this as I can't replicate it.

**Comment by lemmi** [comment]
`6.0.1` is out. Tested this briefly tested on `x86_64-glibc` with old boards.

**Comment by PoroCYon** [comment]
Thanks for the notice, I've update the templates to 6.0.1. `kicad-doc` and `kicad-i18n` don't seem to have downloads anymore, so I've removed the packages.

File: srcpkgs/kicad/template [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-configure_args="-DKICAD_BUILD_VERSION=${version} -DKICAD_SCRIPTING=ON
- -DKICAD_SCRIPTING_MODULES=ON -DKICAD_SCRIPTING_WXPYTHON=ON
- -DKICAD_SCRIPTING_ACTION_MENU=ON  -DBUILD_GITHUB_PLUGIN=ON -DKICAD_USE_OCE=OFF
+configure_args="-DKICAD_SCRIPTING=ON -DKICAD_SCRIPTING_MODULES=ON
+ -DKICAD_SCRIPTING_WXPYTHON=ON -DKICAD_SCRIPTING_ACTION_MENU=ON
  -DKICAD_SCRIPTING_PYTHON3=ON -DKICAD_SCRIPTING_WXPYTHON_PHOENIX=ON
  -DwxWidgets_CONFIG_EXECUTABLE=$WX_CONFIG
  -DKICAD_USE_OCC=$(vopt_if occt ON OFF) -DKICAD_SPICE=$(vopt_if spice ON OFF)"
-hostmakedepends="pkg-config swig wxWidgets-gtk3-devel python3"
+hostmakedepends="pkg-config swig wxWidgets-gtk3-devel python3 tar"
```
Comment (ericonr) [inline srcpkgs/kicad/template:12]: What does it need tar for?

Just want to be sure it isn't downloading something during the build

File: srcpkgs/kicad/template [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 makedepends="wxWidgets-gtk3-devel wxPython4 python3-devel glew-devel cairo-devel
- openssl-devel boost-devel libcurl-devel glm libgomp-devel
+ openssl-devel boost-devel libcurl-devel glm libgomp-devel libfreeglut-devel
```
Comment (ericonr) [inline srcpkgs/kicad/template:14]: I think we managed to remove the dependency on OpenSSL from kicad 6, can you try removing it from here to see if it builds?

You have `libcurl-devel` repeated here.

File: srcpkgs/kicad/template [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+version=6.0.1
+revision=1
 build_style=cmake
 build_helper=cmake-wxWidgets-gtk3
-configure_args="-DKICAD_BUILD_VERSION=${version} -DKICAD_SCRIPTING=ON
- -DKICAD_SCRIPTING_MODULES=ON -DKICAD_SCRIPTING_WXPYTHON=ON
- -DKICAD_SCRIPTING_ACTION_MENU=ON  -DBUILD_GITHUB_PLUGIN=ON -DKICAD_USE_OCE=OFF
+configure_args="-DKICAD_SCRIPTING=ON -DKICAD_SCRIPTING_MODULES=ON
+ -DKICAD_SCRIPTING_WXPYTHON=ON -DKICAD_SCRIPTING_ACTION_MENU=ON
  -DKICAD_SCRIPTING_PYTHON3=ON -DKICAD_SCRIPTING_WXPYTHON_PHOENIX=ON
```
Comment (ericonr) [inline srcpkgs/kicad/template:9]: Kicad always requires python 3 now and phoenix shouldn't have to be specified. I think these options should be removed.

**Review by ericonr — COMMENTED** [review]
These should be separate commits for each template.

`kicad` should get a `replaces="kicad-i18n>=0"` line, since translations are now shipped there instead of in a separate repo.

I'm looking into kicad-doc as well now.

File: srcpkgs/kicad/template (hunk 1 above)
Comment (PoroCYon) [reply srcpkgs/kicad/template:12]: Iirc it errored when trying to unpack the source right after I had done an `xbps-src zap && xbps-src binary-bootstrap`, though I might be wrong, it's been more than a week.

What would it be downloading, though?

**Comment by PoroCYon** [comment]
I've split the commits into separate ones for all the packages, and integrated your feedback.

File: srcpkgs/kicad/template (hunk 1 above)
Comment (ericonr) [reply srcpkgs/kicad/template:10]: That checks out, I had forgotten about the new requirement.

**Comment by PoroCYon** [comment]
Do you have any news on the kicad-doc situation?

**Comment by foopub** [comment]
> Do you have any news on the kicad-doc situation?

What's the issue? https://kicad-downloads.s3.cern.ch/docs/kicad-doc-6.0.1.tar.gz works for me.

**Comment by PoroCYon** [comment]
Ah, the 6.0.0 link was 404'ing. I'll readd the package tomorrow.

**Comment by ericonr** [comment]
For next time, it'd be best to just let CI run. See bb4b6fb5588eeb2dce6611b89c542f082bd14120

**Comment by foopub** [comment]
> For next time, it'd be best to just let CI run. See [bb4b6fb](https://github.com/void-linux/void-packages/commit/bb4b6fb5588eeb2dce6611b89c542f082bd14120)

Seems CI builds fail for all but x86_64... https://github.com/void-linux/void-packages/actions/runs/1736427514
I tried looking through the build logs but they're too long to view online. Tried to download, but it reached >100mb before I cancelled. Not sure what the issues are, and kicad is the most annoying program to debug in my experience (no runtime logs!) so I gave up at that point.

**Pull request closed without merging**